src/js/drop-down.js

```diff
@@ -45,28 +45,47 @@ class DropDown {
 	 * @return {void}
 	 */
 	handleEvent(event) {
-		if (event.type === 'click') {
-			if (!event.target.parentNode || !event.target.parentNode.getAttribute('data-o-header-services-level')) {
+		if (event.key === 'Escape') {
+			this.reset();
+		}
+
+		if (event.type === 'click' && event.target) {
+			// Close dropdown if some non-nav element on the page is clicked.
+			if (event.target.nodeName !== 'BUTTON' &&
+				event.target.nodeName !== 'A' &&
+				event.target !== this.drawer.navList
+			) {
 				this.reset();
 				return;
 			}
 
-			const target = event.target.closest('li');
-			if (!DropDown.isExpanded(target) && event.target.type === 'button') {
+			// Bail if there's no parent menu to toggle. Note: IE11 does not
+			// support `Array.prototype.find()` or `Element.closest()`,
+			// adding a polyfill requirement is a breaking change
+			let parentMenu;
+			for (const navItem of this.navItems) {
+				if (navItem.contains(event.target)) {
+					parentMenu = navItem;
+					break; // found the parent menu
+				}
+			}
```

Comment:
```suggestion
			let parentMenu = this.navItems.find(item => item.contains(event.target));
```

Reply:
`find` would require a new polyfill to support IE11, which we can't introduce without a breaking change. :(

Reply:
oh i completely missed that ^_^, let me try again